FrontEnd/src/App.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import './index.css';
+import Recaptcha from 'react-recaptcha';
+
+export default class App extends Component {
+  render() {
+    return (
+      <Register />
+    );
+  }
+}
+
+// eslint-disable-next-line
+const validEmailRegex = RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
+
+const validateForm = (errors) => {
+    let valid = true;
+    Object.values(errors).forEach(
+      (val) => val.length > 0 && (valid = false)
+    );    
+    return valid;
+}
+
+class Register extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.recaptchaLoaded = this.recaptchaLoaded.bind(this);
+    this.verifyCallback = this.verifyCallback.bind(this);
```

Comment:
Could you explain why this is necessary? Specifically binding `this` to the functions.

Reply:
I can not. 

An example I followed was doing this and since it worked, I went with it.  Unfortunately, I blindly accepted it ( Went into this with zero JS/React experience, so I just simply didn't know any better )

Reply:
It needs to be done when we have something like this:
```jsx
class Foo {
  doSomething() {}
  func() {
    this.doSomething();
  }

  render() {
    <SomeComponent func={this.func} />
  }
}
```

When you have a function declared like that, `this` refers to the `this` inside of the function, rather than the instance of the class. So, that binding syntax is used to override that so you can refer to the instance of the class rather than the function that's passed into the component. 

This can be avoided if you instead declare your functions in classes as arrow functions like this:
```js
  func = () => {
    this.doSomething()
  }
```
because arrow functions maintain the lexical scope of `this`, rather than the block scope of `this` in functions.